### PR TITLE
[fix] Move _adjust_logits above postprocess to fix Marian.generate

### LINKED ISSUE #5124: SLOW GPU test Failures
```bash
=========================== short test summary info ============================
FAILED tests/test_modeling_marian.py::TestMarian_RU_FR::test_batch_generation_ru_fr
FAILED tests/test_modeling_marian.py::TestMarian_MT_EN::test_batch_generation_mt_en
==== 2 failed, 1110 passed, 392 skipped, 361 warnings in 1347.28s (0:22:27) ====
```

https://github.com/huggingface/transformers/runs/782480196?check_suite_focus=true

### PR DESCRIPTION
**Pull request by sshleifer**
Fixes slow gpu test failures caused by #5031 

This logic was already discussed in a previous PR, but, to summarize, the marian model loves to predict `pad_token_id` and if we allow pad_token_id's high score to enter the softmax we get very low probabilities for everything else.

I also rename `prepare_logits_for_generation` -> `_adjust_logits`

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Don't really think that `_adjust_logits` is a better name. Also why the underscore `_` ? we don't do this for `prepare_input_ids_for_generation()` either

**Comment by sshleifer** [comment]
Because it's not user facing, but I guess that's inconsistent. What do you think about `adjust_logits`?

**Comment by patrickvonplaten** [comment]
Hmm, I like `prepare_logits_for_generation` better than `adjust_logits`,  since it has the name `generation` in it, so people know that this function only belongs to generation in bart & marian

**Review by LysandreJik — APPROVED** [review]
